Objects/typeobject.c

```diff
@@ -8134,225 +8127,240 @@ typedef struct wrapperbase slotdef;
-#define MPSLOT(NAME, SLOT, FUNCTION, WRAPPER, DOC) \
-    ETSLOT(NAME, as_mapping.SLOT, FUNCTION, WRAPPER, DOC)
-#define NBSLOT(NAME, SLOT, FUNCTION, WRAPPER, DOC) \
-    ETSLOT(NAME, as_number.SLOT, FUNCTION, WRAPPER, DOC)
-#define UNSLOT(NAME, SLOT, FUNCTION, WRAPPER, DOC) \
+     PyDoc_STR(DOC), .name_strobj = NAMEOBJ}
+#define AMSLOT(NAME, SLOT, FUNCTION, WRAPPER, DOC, NAMEOBJ) \
+    ETSLOT(NAME, as_async.SLOT, FUNCTION, WRAPPER, DOC, NAMEOBJ)
+#define SQSLOT(NAME, SLOT, FUNCTION, WRAPPER, DOC, NAMEOBJ) \
+    ETSLOT(NAME, as_sequence.SLOT, FUNCTION, WRAPPER, DOC, NAMEOBJ)
+#define MPSLOT(NAME, SLOT, FUNCTION, WRAPPER, DOC, NAMEOBJ) \
+    ETSLOT(NAME, as_mapping.SLOT, FUNCTION, WRAPPER, DOC, NAMEOBJ)
+#define NBSLOT(NAME, SLOT, FUNCTION, WRAPPER, DOC, NAMEOBJ) \
+    ETSLOT(NAME, as_number.SLOT, FUNCTION, WRAPPER, DOC, NAMEOBJ)
+#define UNSLOT(NAME, SLOT, FUNCTION, WRAPPER, DOC, NAMEOBJ) \
     ETSLOT(NAME, as_number.SLOT, FUNCTION, WRAPPER, \
-           NAME "($self, /)\n--\n\n" DOC)
-#define IBSLOT(NAME, SLOT, FUNCTION, WRAPPER, DOC) \
+           NAME "($self, /)\n--\n\n" DOC, NAMEOBJ)
+#define IBSLOT(NAME, SLOT, FUNCTION, WRAPPER, DOC, NAMEOBJ) \
     ETSLOT(NAME, as_number.SLOT, FUNCTION, WRAPPER, \
-           NAME "($self, value, /)\n--\n\nReturn self" DOC "value.")
-#define BINSLOT(NAME, SLOT, FUNCTION, DOC) \
+           NAME "($self, value, /)\n--\n\nReturn self" DOC "value.", \
+           NAMEOBJ)
+#define BINSLOT(NAME, SLOT, FUNCTION, DOC, NAMEOBJ) \
     ETSLOT(NAME, as_number.SLOT, FUNCTION, wrap_binaryfunc_l, \
-           NAME "($self, value, /)\n--\n\nReturn self" DOC "value.")
-#define RBINSLOT(NAME, SLOT, FUNCTION, DOC) \
+           NAME "($self, value, /)\n--\n\nReturn self" DOC "value.", \
+           NAMEOBJ)
+#define RBINSLOT(NAME, SLOT, FUNCTION, DOC, NAMEOBJ) \
     ETSLOT(NAME, as_number.SLOT, FUNCTION, wrap_binaryfunc_r, \
-           NAME "($self, value, /)\n--\n\nReturn value" DOC "self.")
-#define BINSLOTNOTINFIX(NAME, SLOT, FUNCTION, DOC) \
+           NAME "($self, value, /)\n--\n\nReturn value" DOC "self.", \
+           NAMEOBJ)
+#define BINSLOTNOTINFIX(NAME, SLOT, FUNCTION, DOC, NAMEOBJ) \
     ETSLOT(NAME, as_number.SLOT, FUNCTION, wrap_binaryfunc_l, \
-           NAME "($self, value, /)\n--\n\n" DOC)
-#define RBINSLOTNOTINFIX(NAME, SLOT, FUNCTION, DOC) \
+           NAME "($self, value, /)\n--\n\n" DOC, NAMEOBJ)
+#define RBINSLOTNOTINFIX(NAME, SLOT, FUNCTION, DOC, NAMEOBJ) \
     ETSLOT(NAME, as_number.SLOT, FUNCTION, wrap_binaryfunc_r, \
-           NAME "($self, value, /)\n--\n\n" DOC)
+           NAME "($self, value, /)\n--\n\n" DOC, NAMEOBJ)
 
 static slotdef slotdefs[] = {
-    TPSLOT("__getattribute__", tp_getattr, NULL, NULL, ""),
-    TPSLOT("__getattr__", tp_getattr, NULL, NULL, ""),
-    TPSLOT("__setattr__", tp_setattr, NULL, NULL, ""),
-    TPSLOT("__delattr__", tp_setattr, NULL, NULL, ""),
+    TPSLOT("__getattribute__", tp_getattr, NULL, NULL, "", &_Py_ID(__getattribute__)),
```

Comment:
```suggestion
    TPSLOT(__getattribute__, tp_getattr, NULL, NULL, ""),
```

Reply:
I am not sure about this, the regex currently searches for `_Py_ID(...)` so it will not pick this up.

Reply:
Yeah, see https://github.com/python/cpython/pull/94706#discussion_r917292060.